### PR TITLE
Add cookiecutter template to create new libraries

### PR DESCRIPTION
**Pull request by smelc**
## What this PR does

* This PR adds a cookiecutter template to create new libraries
* It augments the CI so that the template is called and generated files are checked like existing files

## How to trust this PR

Install cookiecutter in the repo:

```
python3 -m venv .venv
source .venv/bin/activate
pip install -r pip-requirements.txt
pip install -r dev-requirements.txt
```

Then generate a new library:

```
cookiecutter --no-input templates/pylibrary --output-dir libs/
```

Check that the library's CI passes:

```
act workflow_dispatch -j libs-library_name-ci
```

Check that the generated library is consistent with existing libraries:

```
meld libs/base libs/library_name
```

* Inspect that the CI on a generated library is fine, see https://github.com/tweag/python-monorepo-example/pull/5
* Install `libs/base` or `libs/fancy` locally using the new `README.md` and witness the instructions work fine.

## Notes

* I will make the tests within `base` and `fancy` meaningful in another PR.
* I will maybe add CODEOWNERS augmentation in the template in another PR.

### CODE REVIEW

**Comment by smelc** [comment]
@GuillaumeDesforges> please don't merge right away after your review, I'd like to debug the Mergify configuration with this PR. I think it's not working.

**Review by GuillaumeDesforges — APPROVED** [review]

**Comment by smelc** [comment]
Since it's taking forever to get mergify support added to this repo, merging.

**Pull request merged**
